vms/platformvm/validators/manager.go

```diff
@@ -237,7 +248,17 @@ func (m *manager) getValidatorSetCache(subnetID ids.ID) cache.Cacher[uint64, map
 		return &cache.Empty[uint64, map[ids.NodeID]*validators.GetValidatorOutput]{}
 	}
 
+	m.cachesLock.RLock()
```

Comment:
can't we a single
```
        m.cachesLock.Lock()
	defer m.cachesLock.Unlock()
```
block here? After all if `exists` is true we return immediately

Reply:
We could - this is just a way to avoid unnecessary locking.

Reply:
Oh wait. I missed the first is Rlock. Fine as it is.